### DISCUSSION TITLE
Could not load the Qt platform plugin "offscreen"

### DISCUSSION THREAD

**Discussion by gianmore** [rajewsky-lab/openst#65] [Q&A]
Hi, when at this point of your tutorial https://rajewsky-lab.github.io/openst/main/examples/e13_mouse/pairwise_alignment/#fine-pairwise-alignment-manual I get this error: INFO 2024-10-02 15:33:25,242 - openst from_spacemake - running from spacemake directory
INFO 2024-10-02 15:33:25,242 - {'subcommand': 'from_spacemake', 'project_id': 'openst_demo', 'sample_id': 'openst_demo_e13_mouse_head', 'run_mode': '', 'func': <function cmd_run_from_spacemake at 0x7fe058ed6d40>}
INFO 2024-10-02 15:33:26,425 - openst manual_pairwise_aligner - running with the following parameters:
INFO 2024-10-02 15:33:26,425 - {'project_id': 'openst_demo', 'sample_id': 'openst_demo_e13_mouse_head', 'run_mode': '', 'subcommand': 'manual_pairwise_aligner', 'h5_in': 'projects/openst_demo/processed_data/openst_demo_e13_mouse_head/multimodal/stitched_spots.h5ad', 'spatial_key': 'obsm/spatial_pairwise_aligned_coarse', 'image_key': 'uns/spatial_pairwise_aligned/staining_image_transformed', 'func': <function cmd_run_manual_pairwise_aligner at 0x7fe058ed65c0>}
qt.qpa.plugin: Could not load the Qt platform plugin "offscreen" in "/igatech/rd/dev/mambaforge/envs/openst/lib/python3.11/site-packages/cv2/qt/plugins" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: xcb, eglfs, linuxfb, minimal, minimalegl, offscreen, vnc, wayland-egl, wayland, wayland-xcomposite-egl, wayland-xcomposite-glx, webgl.

8_openst_manual_pairwise_[aligner.sh](http://aligner.sh/): line 8: 29048 Aborted                 openst from_spacemake --project-id openst_demo --sample-id openst_demo_e13_mouse_head manual_pairwise_aligner --spatial-key obsm/spatial_pairwise_aligned_coarse --image-key uns/spatial_pairwise_aligned/staining_image_transformed
 I have tried to unistall and reinstall the packages but it doesn't work. I can see that in my environemnt I have the following installed:                                                                                         pyqt                          5.15.9          py311hf0fb5b6_5    conda-forge
pyqt5-qt5                 5.15.2                   pypi_0    pypi
pyqt5-sip                 12.12.2                  pypi_0    pypi
pyqtgraph                 0.13.7                   pypi_0    pypi                                       is it an issue with the versions of pyqt? which one do you have in your openst environment? what else can I check? thanks!

**Comment by danilexn** [comment]
Hi! Replied on chat: 

This error is a conflict between PyQt and opencv-Python that has been documented:

> In some environments, the following error might happen:
```bash
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "/home/user/miniconda3/envs/openst/lib/python3.11/site-packages/cv2/qt/plugins" even though it was found.

This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: xcb, eglfs, linuxfb, minimal, minimalegl, offscreen, vnc, wayland-egl, wayland, wayland-xcomposite-egl, wayland-xcomposite-glx, webgl.
```

This can be solved by removing the files inside e.g. `/home/user/miniconda3/envs/openst/lib/python3.11/site-packages/cv2/qt/plugins`, e.g. see this StackOverflow thread: https://stackoverflow.com/questions/71029300/python-pyqt-xcb-plugin-with-thread-error

**Reply by gianmore** [reply]
It didn't solve the issue for me. can you please tell me the versions that you are using in your environment of opencv-python, PyQt5 and related packages? Thanks
